services/src/main/java/org/keycloak/services/DefaultKeycloakSessionFactory.java

```diff
@@ -229,6 +217,10 @@ private boolean isEnabled(ProviderFactory factory, Config.Scope scope) {
         if (factory instanceof EnvironmentDependentProviderFactory) {
             return ((EnvironmentDependentProviderFactory) factory).isSupported();
         }
+        if (factory instanceof FeatureDependentProviderFactory) {
+            return Optional.ofNullable(((FeatureDependentProviderFactory) factory).getRequiredFeatures()).orElse(Collections.emptySet())
+                    .stream().allMatch(requiredFeature -> Profile.isFeatureEnabled(requiredFeature));
+        }
```

Comment:
Yes. I agree conditional would be a much better name. However, renaming it breaks backwards compatibility so that's not an option and I don't like alternatives so it is what it is ;)